src/include/sof/audio/volume.h

```diff
@@ -146,6 +148,7 @@ struct vol_data {
 	int32_t vol_min;			/**< minimum volume */
 	int32_t vol_max;			/**< maximum volume */
 	int32_t	vol_ramp_range;			/**< max ramp transition */
+	int32_t sample_rate_inv;		/**< 1000x inverse of sample rate as Q1.31 */
```

Comment:
better find another way to set ramp time instead of add new var to do calculation?
any concerns to set ramp time with a fixed or a table with different value, like 8k to 100ms, 48k use 20ms? just a example.

Reply:
The rate inverse coefficient is calculated only once in prepare(). After that one in ramp update the sample counter is multiplied by the coefficient. The ramp functions need time, so I don't see how this part could be made any simpler with lookup table. The ramp length is arbitrary with 1 ms steps. I'd not like to pre-define some ramp times.